compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -397,16 +398,25 @@ private void addRequiredParamsToGeneratedInitFunction(BLangFunction initFunction
         if (initFunction.requiredParams.isEmpty()) {
             return;
         }
+
+        SymbolEnv preEnv = this.env;
+        this.env = SymbolEnv.createFunctionEnv(generatedInitFunc, generatedInitFunc.symbol.scope, env);
```

Comment:
Should we consider overloading the `rewriteExpr()` method to take an env as an argument and make this env switching logic a part of that API? We haven't done this anywhere else and if we start doing it, could potentially become a mess.

@hasithaa thoughts?

Reply:
I think passing an env to rewriteExpr is not a common thing.  We usually create env when we visit the node.  This particular scenario is a special case, where the same node repeating in two different parent nodes. Do you think it will be a common case? If so I am +1 for overloading.

Reply:
Not sure if it'd become a common thing. My concern was if people see this usage and start to do the same in other places.